docs/public/components/faq.md

```diff
@@ -0,0 +1,38 @@
+# FAQ
+
+TODO: Should we migrate the FAQ from confluence here ?
+
+## Optional dependency
+
+You might need to express the fact that some of your dependencies are optional. This often happens for
+components that interact with many other components **if available** (that is, if they were included at compile time). This allows
```

Comment:
This happens when a component can be enabled or not at runtime during the creation of the component. This component also have to expose methods.